### DISCUSSION TITLE
$GITHUB_STEP_SUMMARY - how to append a file with markdown contents?

### DISCUSSION THREAD

**Discussion by ignatandrei** [community/community#26661] [Actions]
<p>I have a github action that run tests and the output is a BDD summary<br>
I want to append this on the $GITHUB_STEP_SUMMARY  - but the markdown is rendered as it was on a ingle line</p>
<p>Here is how I do:<br>
( code at <a href="https://github.com/ignatandrei/TILT/blob/be80182145f241119940f513d996d32fd3e72f0f/.github/workflows/build.yml" class="inline-onebox" rel="noopener nofollow ugc">TILT/build.yml at be80182145f241119940f513d996d32fd3e72f0f · ignatandrei/TILT · GitHub</a>)</p>
<ul>
<li>name: premark<br>
shell: pwsh<br>
run: |<br>
cd src/backend/Net6/NetTilt/NetTilt.Tests<br>
dir bin/Debug/net6.0/Reports/*.md<br>
$test=(Get-Content bin/Debug/net6.0/Reports/LightBDDReport.md)<br>
$test = $test -replace ‘"’,’’’’<br>
echo “action_state=yellow”  | Out-File -FilePath $Env:GITHUB_ENV -Encoding utf-8 -Append<br>
echo “test_mkd=$test”  | Out-File -FilePath $Env:GITHUB_ENV -Encoding utf-8 -Append
<ul>
<li>name: mark<br>
run: |<br>
echo “### Hello world! :rocket:” &gt;&gt; $GITHUB_STEP_SUMMARY<br>
echo “This is the lead in sentence for the list” &gt;&gt; $GITHUB_STEP_SUMMARY<br>
echo “” &gt;&gt; $GITHUB_STEP_SUMMARY # this is a blank line<br>
echo “- Lets add a bullet point” &gt;&gt; $GITHUB_STEP_SUMMARY<br>
echo “- Lets add a second bullet point” &gt;&gt; $GITHUB_STEP_SUMMARY<br>
echo “- How about a third one?” &gt;&gt; $GITHUB_STEP_SUMMARY<br>
echo “${{ env.action_state }}” &gt;&gt; $GITHUB_STEP_SUMMARY<br>
echo “${{ env.test_mkd }}” &gt;&gt; $GITHUB_STEP_SUMMARY<br>
echo “### Hello world! :rocket:” &gt;&gt; $GITHUB_STEP_SUMMARY</li>
</ul>
</li>
</ul>
<p>( see the output at <a href="https://github.com/ignatandrei/TILT/actions/workflows/build.yml" class="inline-onebox" rel="noopener nofollow ugc">Actions · ignatandrei/TILT · GitHub</a> )</p>
<p>Please help</p>

**Comment by orta** [comment]
<p>the unix <code>&gt;&gt;</code> operator pipes in the prior stdout as-is, you’d need to add <code>'\n'</code> to each line you are adding this way:</p>
<p>e.g.</p>
<pre><code class="lang-auto">echo “### Hello world! :rocket:\n” &gt;&gt; $GITHUB_STEP_SUMMARY
</code></pre>
<p>I expect.</p>

**Comment by ignatandrei** [comment]
<p>I have a file that contains markdown. How I can put the content of the  file into $GITHUB_STEP_SUMMARY  ?</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>Just <code>cat</code> the file and redirect the output, like so:</p>
<pre><code class="lang-auto">cat FILE &gt;$GITHUB_STEP_SUMMARY
</code></pre>
<p>Use <code>&gt;&gt;</code> instead of <code>&gt;</code> if you want to append to instead of overwrite the step summary file.</p>

**Comment by ignatandrei** [comment]
<p>Thanks, it works perfectly ( forgot about cat )</p>
